tests/sles4sap/saptune/mr_test.pm

```diff
@@ -17,6 +17,19 @@ use version_utils 'is_sle';
 use strict;
 use warnings;
 
+sub reboot_wait {
+    my ($self) = @_;
+
+    $self->reboot;
+
+    # Wait for tuned to tune everything
+    if (is_sle('>=15')) {
+        assert_script_run "timeout 60 bash -c 'until tuned-adm verify >/dev/null ; do sleep 1 ; done'", 60;
```

Comment:
`tuned-adm verify` doesn't work on SLE12 (didn't try)?

Reply:
I'm also not so happy with `[...] timeout 60 bash -c '[...]'", 60;` (and in the `sleep 60` also). Maybe a variable can be useful here?

I know that this shell code works, but maybe better to use a more "Perlish" syntax like this (we already have some code like this in `lib/sles4sap.pm`):

    my $time_to_wait = 60;
    my $sleep_time   = 5;
    while ($time_to_wait > 0) {
        my $ret = script_run('tuned-adm verify >/dev/null');
        last if (defined $ret and $ret == 0);

        $time_to_wait -= $sleep_time;
        sleep $sleep_time;
    }

Reply:
> `tuned-adm verify` doesn't work on SLE12 (didn't try)?

No.

Reply:
> I'm also not so happy with `[...] timeout 60 bash -c '[...]'", 60;` (and in the `sleep 60` also). Maybe a variable can be useful here?

We already mix Bash & Perl.  I prefer the one-liner Bash solution because it resembles what I have on the mr_tester.sh Bash script for testing outside of openQA.

Reply:
> I prefer this one-liner Bash solution because it resembles what I have on the mr_tester.sh Bash script for testing outside of openQA.

If @alvarocarvajald is also OK that's fine for me, but please just use a variable for the timeout value :D

Reply:
> `tuned-adm verify` doesn't work on SLE12 (didn't try)?

It doesn't:

```
1a118:~ # tuned-adm
usage: tuned-adm [-h] [--version] [--debug]
                 {list,active,off,profile,recommend} ...
tuned-adm: error: too few arguments
```
At least with tuned-2.4.1-16.26.noarch.

Agree that a variable for the timeout would be nice. And also add a `bmwqemu::scale_timeout()` to the sleep (assert_script_run will automatically scale the timeout if TIMEOUT_SCALE is set)

Reply:
> Agree that a variable for the timeout would be nice. And also add a `bmwqemu::scale_timeout()` to the sleep (assert_script_run will automatically scale the timeout if TIMEOUT_SCALE is set)

We can add a variable but it wouldn't be useful as it would mask that bug.  Let's see if the developers care about making user sessions available only after tuned has finished configuring the system.

Reply:
Sounds logical. Just keep in mind that the 60 seconds in the `assert_script_run()` call and the 60 seconds in the `sleep` call will not be the same if TIMEOUT_SCALE is set to something above 1.